app/controllers/events/view_transcript.py

```diff
@@ -0,0 +1,70 @@
+from app.models.course import Course
+from app.models.courseParticipant import CourseParticipant
+from app.models.program import Program
+from app.models.courseInstructors import CourseInstructors
+from app.models.user import User
+from app.models.term import Term
+from app.models.eventParticipant import EventParticipant
+from app.models.event import Event
+
+
+def getSLCourseTranscript(user):

```

Comment:
This whole function is almost a textbook candidate for a single SQL query. Maybe two, if you wanted to get instructors separate from hours, but I would look into SQL aggregate functions and GROUP BY, or what the peewee version of that would be. Then you wouldn't have to do any of the loops and conditionals

Reply:
And the nice thing about having a test for it (good job), is you've got an easy way to keep running it and verify that the results are correct.

Reply:
Our new function is using a query to get all the necessary information for the course trancript. We used a loop to append the information into a list of dictionaries.